src/installer/managed/Microsoft.NET.HostModel/Bundle/Manifest.cs

```diff
@@ -93,38 +87,9 @@ public long Write(BinaryWriter writer)
             return startOffset;
         }
 
-        public static Manifest Read(BinaryReader reader, long headerOffset)
+        public bool Contains(string relativePath)
         {
-            // Read the bundle header
-            reader.BaseStream.Position = headerOffset;
-            uint majorVersion = reader.ReadUInt32();
-            uint minorVersion = reader.ReadUInt32();
-            int fileCount = reader.ReadInt32();
-            string bundleID = reader.ReadString();
-
-            bool isCompatible = 
-                (majorVersion < MajorVersion) ||
-                (majorVersion == MajorVersion && minorVersion <= MinorVersion);
-
-            if (!isCompatible)
-            {
-                throw new BundleException("Extraction failed: Invalid Version");
-            }
-
-            // Read the manifest entries
-            Manifest manifest = new Manifest(bundleID);
-            for (long i = 0; i < fileCount; i++)
-            {
-                manifest.Files.Add(FileEntry.Read(reader));
-            }
-
-            if (manifest.Files.GroupBy(file => file.RelativePath).Where(g => g.Count() > 1).Any())
-            {
-                throw new BundleException("Extraction failed: Found multiple entries with the same bundle-relative-path");
-            }
-
-            return manifest;
+            return Files.Any(entry => relativePath.Equals(entry.RelativePath));
```

Comment:
~~I worry that this could induce accidentally quadratic behavior.  Maybe change the type of Files to be `HashSet<FileEntry>`?~~  (Nevermind, this seems to be used only for the testing.)